git/git.go

```diff
@@ -727,6 +738,54 @@ func (git *Git) ListUncommitted(dirs ...string) ([]string, error) {
 	return removeEmptyLines(strings.Split(out, "\n")), nil
 }
 
+// ShowCommitMetadata returns common metadata associated with the given object.
+// An object name can be a commit SHA or a symbolic name, i.e. HEAD, branch-name, etc.
+func (git *Git) ShowCommitMetadata(objectName string) (*CommitMetadata, error) {
+	// %n - newline
+	// %an - author name
+	// %at - author time (unix)
+	// %ae - author email
+	// %s - commit msg subject
+	// %b - commit msg body
+	out, err := git.exec("show", "-s", "--format=%an%n%at%n%ae%n%s%n%b", objectName)
+	if err != nil {
+		return nil, fmt.Errorf("show: %w", err)
+	}
+
+	lines := strings.SplitN(out, "\n", 5)
+	// 4 lines without body, 5+ with body
+	if len(lines) < 4 {
+		return nil, fmt.Errorf("show metadata: malformed output: %v", out)
+	}
+
+	author := strings.TrimSpace(lines[0])
+
+	var commitTime *time.Time
+	unixTime, err := strconv.ParseInt(strings.TrimSpace(lines[1]), 10, 64)
+	if err == nil {
+		v := time.Unix(unixTime, 0)
+		commitTime = &v
+	} else {
+		commitTime = nil
+	}
+
+	email := strings.TrimSpace(lines[2])
+	subject := strings.TrimSpace(lines[3])
+
+	body := ""
+	if len(lines) > 4 {
+		body = strings.TrimSpace(lines[4])
```

Comment:
Isn't this just getting the first line of the commit body?

Reply:
I'll add a test for this as suggested, but this works since I used strings.SplitN, not .Split, so lines[4] contains the remaining unsplit body.